### PR TITLE
Initial Checkin - Content Pack - Commvault Security IQ

### PR DESCRIPTION
**Pull request by Cv-securityIQ**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by Cv-securityIQ** [comment]
@RotemAmit with whom should I share the video for the Contribution Registration? I would need an email ID to provide access to the video.

**Comment by RotemAmit** [comment]
> @RotemAmit with whom should I share the video for the Contribution Registration? I would need an email ID to provide access to the video.

Hi @Cv-securityIQ, you can share the video with me. I am available in DFIR as well.
Please fill the [CLA](https://cla-assistant.io/demisto/content?pullRequest=27236) form for @amitshakya-cv as well.

**Review by efelmandar — COMMENTED** [review]
@Cv-securityIQ Thank you very much for your contribution, I finished reviewing the PR and overall it looks excellent, but a couple of points I do want to bring up are:

1. For all content items (incident types, fields, mapper, playbooks, and layout) please make sure to prefix the name with the pack name.

2. In the playbook, the `Disabling data aging` task returns a status response for the action, why not reflect it to the user or at least in the closing notes?

Please let me know if you have any further questions or need any assistance,

**Comment by Cv-securityIQ** [comment]
We are currently making changes based on code review comments.

**Comment by Cv-securityIQ** [comment]
> @Cv-securityIQ Thank you very much for your contribution, I finished reviewing the PR and overall it looks excellent, but a couple of points I do want to bring up are:
> 
> 1. For all content items (incident types, fields, mapper, playbooks, and layout) please make sure to prefix the name with the pack name.
> 2. In the playbook, the `Disabling data aging` task returns a status response for the action, why not reflect it to the user or at least in the closing notes?
> 
> Please let me know if you have any further questions or need any assistance,

We have another product called Metallic, for which we are intending to use the same integration. If we add Commvault Security in the prefixes it might cause some confusion later on.

**Comment by maimorag** [comment]
> ecause we would prefer it if the users used tokens, instead of username password.

Please notice that "type:9" also can be utilized as a single parameter:
You can use a single parameter, as demonstrated in the following YAML example. In this case, the username is hidden, and the "Token" can be obtained by accessing -
```token = demisto.params().get("credentials", {}).get("password")```

- displaypassword: Token
  name:  credentials
  required: true
  hiddenusername: true
  type: 9

For more information regarding types you can check the integration "ipinfo_v2" line 9.

**Comment by Cv-securityIQ** [comment]
We've changed the parameter type for the tokens (CommvaultAPIToken,AzureKeyVaultClientSecret) to type 9, and also added the company name as a prefix to all the custom fields, layout, incident type, playbook and mapper. Can you please review it  ?

**Comment by edik24** [comment]
Hi @Cv-securityIQ,
please see [here](https://xsoar.pan.dev/docs/concepts/design-best-practices) information about command names.

It should be:
PRODUCTNAME-OBJECTNAME-ACTION

**Comment by Cv-securityIQ** [comment]
> Hi @Cv-securityIQ, please see [here](https://xsoar.pan.dev/docs/concepts/design-best-practices) information about command names.
> 
> It should be: PRODUCTNAME-OBJECTNAME-ACTION

Hi @edik24 , noted. We've changed the command names.

**Comment by Cv-securityIQ** [comment]
@edik24 All the Review comments are handled and the pull request is updated.

**Comment by efelmandar** [comment]
Hi @Cv-securityIQ I see you made the changes we discussed and it looks good, but I see you created an incident field named `Event ID` which already exists in the CommonTypes pack, please use the existing one as we try to minimize duplication of objects in the system

**Comment by Cv-securityIQ** [comment]
Hi @efelmandar , @maimorag made the changes you suggested.

**Comment by maimorag** [comment]
Few notes:
Unit tests must be present for all integrations/scripts commands (including fetch_incidents) and helper functions.
Missing unit tests( besides clients’ functions):
set_secret_in_key_vault
long-running-execution
fetch-incidents
field_mapper
2. please update the README - please run the `demisto-sdk generate-docs  -c  commvault-security-get-access-token-from-keyvault ` command (also, use the -e flag for using a commad_examples and add example for commvault-security-get-access-token-from coammand ) - for more information please see: https://xsoar.pan.dev/docs/concepts/demisto-sdk#generate-docs

**Comment by Cv-securityIQ** [comment]
@maimorag , can you please review the changes ?

**Comment by maimorag** [comment]
Please revert the changes in file .circleci/config.yml

**Comment by Cv-securityIQ** [comment]
> Please revert the changes in file .circleci/config.yml

Do we need to make a change over here? Or Palo Alto team is making these changes internally?

**Comment by maimorag** [comment]
Hi @Cv-securityIQ, the code looks good!

We're ready for a demo. Please check [this page](https://xsoar.pan.dev/docs/contributing/demo-prep), and let me and @edik24 know when you're available for one over DFIR.

File: Packs/CommvaultSecurityIQ/Integrations/CommvaultSecurityIQ/CommvaultSecurityIQ.py [hunk 1]
```diff
@@ -0,0 +1,1473 @@
...
+                client,
+                is_valid_cv_token,
+                is_fetch,
+                is_long_running,
+                forwarding_rule_type,
+            )
+            return_results("ok")
+        elif command == "fetch-incidents":
+            last_fetch, out = fetch_incidents(
+                client, last_run=demisto.getLastRun(), first_fetch_time=first_fetch_time
```
Comment (maimorag) [inline Packs/CommvaultSecurityIQ/Integrations/CommvaultSecurityIQ/CommvaultSecurityIQ.py:1404]:
```suggestion
                client, last_run=demisto.getLastRun().get('last_fetch'), first_fetch_time=first_fetch_time
```

**Review by maimorag — APPROVED** [review]

**Comment by maimorag** [comment]
@Cv-securityIQ , We noticed the forked repo is named **Cv-securityIQ_master**  instead of the suggested 'content' naming convention.
This current situation is preventing us to continue with the merging process.
Could you please manually rename the repo using this manual?
https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository
Please notify me if there are any questions.
Thanks!

**Review by NEEHVN007 — COMMENTED** [review]
Disable pipelines

**Pull request merged**